back/src/Guide/Database/Convert.hs

```diff
@@ -0,0 +1,173 @@
+
+import qualified Data.Map as Map
+import qualified Data.Set as Set
+import qualified Data.Text as T
+import qualified Hasql.Decoders as HD
+import qualified Hasql.Encoders as HE
+
+import Guide.Markdown (toMarkdownBlock, toMarkdownInline, toMarkdownTree)
+import Guide.Types.Core (Category (..), CategoryStatus(..), Item (..), ItemSection(..), Trait (..),
+                         TraitType (..))
+import Guide.Utils (Uid (..))
+
+
+-- Convert functions for database
+
+----------------------------------------------------------------------------
+-- Encoders
+----------------------------------------------------------------------------
+
+-- | nonNullable Bool
+boolEncoder :: HE.NullableOrNot HE.Value Bool
+boolEncoder = HE.nonNullable HE.bool
+
+-- | nullable Bool
+boolEncoderNullable :: HE.NullableOrNot HE.Value (Maybe Bool)
+boolEncoderNullable = HE.nullable HE.bool
+
+-- | nonNullable Text
+textEncoder :: HE.NullableOrNot HE.Value Text
+textEncoder = HE.nonNullable HE.text
+
+-- | nullable Text
+textEncoderNullable :: HE.NullableOrNot HE.Value (Maybe Text)
+textEncoderNullable = HE.nullable HE.text
+
+-- | Encode bool
+boolParam :: HE.Params Bool
+boolParam = HE.param boolEncoder
+
+-- | Encode nullable bool
+boolParamNullable :: HE.Params (Maybe Bool)
+boolParamNullable = HE.param boolEncoderNullable
+
+-- | Encode text
+textParam :: HE.Params Text
+textParam = HE.param textEncoder
+
+-- | Encode nullable text
+textParamNullable :: HE.Params (Maybe Text)
+textParamNullable = HE.param textEncoderNullable
+
+-- | Encode TraitType
+traitTypeParam :: HE.Params TraitType
+traitTypeParam = HE.param (HE.nonNullable $ HE.enum (T.pack . show))
+
+-- | Encode Uid
+uidParam :: HE.Params (Uid a)
+uidParam = contramap uidToText textParam
+
+----------------------------------------------------------------------------
+-- Decoders
+----------------------------------------------------------------------------
+
+-- | nonNullable text
+textDecoder :: HD.NullableOrNot HD.Value Text
+textDecoder = HD.nonNullable HD.text
+
+-- | nullable text
+textDecoderNullable :: HD.NullableOrNot HD.Value (Maybe Text)
+textDecoderNullable = HD.nullable HD.text
+
+-- | Decode text
+textColumn :: HD.Row Text
+textColumn = HD.column textDecoder
+
+-- | Decode nullable text
+textColumnNullable :: HD.Row (Maybe Text)
+textColumnNullable = HD.column textDecoderNullable
+
+-- | Decode [text]
+textArrayColumn :: HD.Row [Text]
+textArrayColumn = (HD.column . HD.nonNullable)
+  $ HD.array (HD.dimension replicateM (HD.element textDecoder))
+
+-- Decode UTCTime
+timestamptzColumn :: HD.Row UTCTime
+timestamptzColumn = HD.column (HD.nonNullable HD.timestamptz)
+
+-- Decode Uid
+uidColumn :: HD.Row (Uid a)
+uidColumn = Uid <$> textColumn
+
+-- | Decode Trait
+traitRow :: HD.Row Trait
+traitRow = do
+  _traitUid <- uidColumn
+  _traitContent <- toMarkdownInline <$> textColumn
+  pure $ Trait{..}
+
+-- | Decode Item
+itemRow :: Text -> [Trait] -> [Trait] -> [Trait] -> [Trait] -> HD.Row Item
```

Comment:
As I said, these functions depend on the order of columns in `SELECT` and therefore they should be in the same place as the `SELECT`.